app/components/provider_interface/safeguarding_declaration_component.rb

```diff
@@ -2,13 +2,41 @@ module ProviderInterface
   class SafeguardingDeclarationComponent < ViewComponent::Base
     include ViewHelper
 
-    attr_reader :message
+    attr_reader :application_choice, :current_provider_user
 
-    def initialize(application_form:)
-      @message = SafeguardingStatus.new(
-        status: application_form.safeguarding_issues_status,
+    def initialize(application_choice:, current_provider_user:)
+      @application_choice = application_choice
+      @current_provider_user = current_provider_user
+    end
+
+    def message
+      SafeguardingStatus.new(
+        status: status,
         i18n_key: 'provider_interface.safeguarding_declaration_component',
       ).message
     end
+
+    def display_safeguarding_information?
```

Comment:
I think this does the right thing but I could see this method name inviting people to put unrelated logic in the body — how about something like `current_user_has_permission_to_view_safeguarding_information?`